SemanticUI/Modules/Dropdown/Select.elm

```diff
@@ -0,0 +1,230 @@
+
+  - identifier - unique identifier
+  - onToggle - message to handle dropdown state changes
+  - toggleEvent - what casuse dropdown state change (default OnClick)
+  - readOnly - is the selection read only or not
+  - onSelect - messages to handle selection change (when clicking on choice)
+  - choices - the list of possible choices
+  - inline - should the selection be inline or not
+    (dropdown does not have "selection" class but "inline" and
+    "text" class is added to the unexpanded item)
+
+-}
+type alias Config r c s msg =
+    { r
+        | identifier : String
+        , onToggle : Dropdown.State -> msg
+        , toggleEvent : Dropdown.ToggleEvent
+        , readOnly : Bool
+        , onSelect : s -> msg
+        , choices : List c
+        , inline : Bool
+        , makeSelectToggle : WrappedNode msg -> Html msg
+    }
+
+
+{-| Create a default Config given, unique identifier, dropdown state change handler
+selection handler and list of choices
+-}
+init :
+    { r
+        | identifier : String
+        , onToggle : Dropdown.State -> msg
+        , onSelect : s -> msg
+        , choices : List c
+    }
+    -> Config {} c s msg
+init { identifier, onToggle, onSelect, choices } =
+    { identifier = identifier
+    , onToggle = onToggle
+    , toggleEvent = Dropdown.OnClick
+    , readOnly = False
+    , onSelect = onSelect
+    , choices = choices
+    , inline = False
+    , makeSelectToggle = \makeSelect -> makeSelect i [ class "dropdown icon" ] []
+    }
+
+{-| Modify a Config's makeSelectToggle value
+-}
+makeSelectToggle :
+    (WrappedNode msg -> Html msg)
+    -> { a | makeSelectToggle : WrappedNode msg -> Html msg }
+    -> { a | makeSelectToggle : WrappedNode msg -> Html msg }
+makeSelectToggle b a =
+    { a | makeSelectToggle = b }
+
+{-| Modify a Config's inline value
+-}
+inline : Bool -> { a | inline : Bool } -> { a | inline : Bool }
+inline b a =
+    { a | inline = b }
+
+
+type alias State r a =
+    { r
+        | dropdownState : Dropdown.State
+        , selectedValue : a
+    }
+
+
+{-| Draw the select.
+
+Takes the following:
+
+  - Configuration,
+  - the current dropdown state,
+  - the currently selected value,
+  - the type of top level DOM node as a render function,
+  - any extra attributes applied to the top level DOM node and
+  - the choice layout function
+
+The choice layout function returns a DOM node and takes:
+
+  - A wrapped node (give it a node render function attributes and children).
+  - A bool that is true choice being rendered is part of the list and false if
+    its the selected choice.
+  - The choice value.
+
+When not inline then "selection" class is added to the top level node otherwise
+"inline" calsss is added and "text" class is added when the choice is not rendered
+as part of the list.
+
+-}
+select :
+    Config r0 a a msg
+    -> State r1 a
+    -> Render msg
+    -> List (Attribute msg)
+    -> (WrappedNode msg -> Bool -> a -> Html msg)
+    -> Html msg
```

Comment:
I feel as though it's a little inconsistent that there's a `Layout` type alias for `dropdown`, but not for `select` or `selectMaybe`. 

Not quite sure what to do about it though...